server/src/test/java/org/apache/uniffle/server/ShuffleTaskManagerTest.java

```diff
@@ -753,7 +807,8 @@ public void checkAndClearLeakShuffleDataTest(@TempDir File tempDir) throws Excep
     assertTrue(appIdsOnDisk.contains(appId));
 
     // make sure heartbeat timeout and resources are removed
-    Thread.sleep(5000);
+    Awaitility.await().timeout(10, TimeUnit.SECONDS).until(
```

Comment:
Why this is involved in this PR?

Reply:
In original logic, `shuffleTaskInfos.remove(appId)` was invoked after remove all resources, but it was invoked before remove all resources in this pr. It becomes a flaky test in this pr.